sqlalchemy_continuum/manager.py

```diff
@@ -490,23 +491,37 @@ def positional_args_to_dict(self, op, statement, params):
 
         :param statement: SQL statement string
         :param params: tuple or dict of statement parameters
+
+        returns: dict with parameters
         """
-        if isinstance(params, tuple):
+
+        if isinstance(params, tuple) and params: # fix for issue raised in #284
+            regex_lookup = {
+                Operation.DELETE:  '^DELETE FROM (.+?) WHERE',
+                Operation.INSERT: '^INSERT INTO (.*?) (VALUES|\()' # INSERT INTO `table_name` VALUES vs INSERT INTO `table_name` (, fix for issue raised in #282
+            }
             parameters = {}
-            if op == Operation.DELETE:
-                regexp = '^DELETE FROM (.+?) WHERE'
+            regexp = regex_lookup.get(op, None)
+            if regexp is not None:
                 match = re.match(regexp, statement)
                 tablename = match.groups()[0].strip('"').strip("'").strip('`')
                 table = self.metadata.tables[tablename]
-                columns = table.primary_key.columns.values()
+                if op == Operation.DELETE:
+                    columns = table.primary_key.columns.values()
+                elif op == Operation.INSERT:
+                    columns = table.columns.values()
                 for index, column in enumerate(columns):
+                    if column.type.__class__.__name__.lower() in ('datetime', ...): # todo add any other type of datetime column?
```

Comment:
When checking for class - prefer using `isinstance(column.type, CLASSNAME)`
In this case - `isinstance(column.type, sa.DateTime)`

But even if you do that - there is no reliable way to get the "type" here. Some DBs use TIMESTAMP, some use DATETIME, TIMETZ, DATE, etc.

Reply:
thanks for the feedback tried to address this in #295